scripts/o3de/o3de/engine_template.py

```diff
@@ -1893,20 +1889,17 @@ def create_gem(gem_path: pathlib.Path,
     # gem restricted name
     if gem_restricted_name and not gem_restricted_path:
         gem_restricted_path = manifest.get_registered(restricted_name=gem_restricted_name)
-
+        if not gem_restricted_path:
+            logger.error(f'Gem Restricted Name {gem_restricted_name} cannot be found.')
+            return 1
     # gem restricted path
     elif gem_restricted_path:
         if not os.path.isabs(gem_restricted_path):
-            default_gems_restricted_folder = manifest.get_registered(restricted_name='gems')
-            new_gem_restricted_path = default_gems_restricted_folder /gem_restricted_path
-            logger.info(f'Gem restricted path {gem_restricted_path} is not a full path, we must assume its'
-                        f' relative to default gems restricted path = {new_gem_restricted_path}')
-            gem_restricted_path = new_gem_restricted_path
-    elif template_restricted_path:
-        gem_restricted_default_path = manifest.get_registered(restricted_name='gems')
-        logger.info(f'--gem-restricted-path is not specified, using default gem restricted path / gem name'
-                    f' = {gem_restricted_default_path}')
-        gem_restricted_path = gem_restricted_default_path
+            logger.error(f'Gem Restricted Path {gem_restricted_path} is not an absolute path.')
```

Comment:
Instead of this being an` if... elif` block to check the gem_restricted_path.
The `elif` should be an if block to validate that the path that `manifest.get_registered` return is actually a path.

Reply:
done